### PR TITLE
Improve language and method selection API

### PR DESCRIPTION
**Pull request by divec**

### CODE REVIEW

**Comment by edg2s** [comment]
not a blocker, but we should update the readme documentation

File: src/jquery.ime.js [hunk 1]
```diff
@@ -48,12 +48,20 @@
 		// This needs to be delayed here since extending language list happens at DOM ready
 		$.ime.defaults.languages = arrayKeys( $.ime.languages );
 		this.options = $.extend( {}, $.ime.defaults, options );
+		if ( this.options.imePath && this.options.imePath !== $.ime.path ) {
+			// Set the global IME path from the one specified to the instance
+			// TODO: remove this functionality and force clients to set the global
+			// IME path
+			$.ime.path = this.options.imePath;
+		}
```
Comment (santhoshtr) [inline src/jquery.ime.js]: Basically we are giving priority for $.options.imePath if defined, but consistantly use $.ime.path in rest of the code. I think in that case the comparison `this.options.imePath !== $.ime.path`  is redundant right?

File: src/jquery.ime.js [hunk 2]
```diff
@@ -73,6 +81,62 @@
...
 
 		/**
+		 * Return a list of available language codes
+		 *
+		 * @return {string[]} Available language codes
+		 */
+		getLanguageCodes: function () {
+			return $.ime.defaults.languages;
+		},
+
```
Comment (santhoshtr) [inline src/jquery.ime.js]: Minor: Multiple line break

**Comment by santhoshtr** [comment]
Looks good to me.  Do you want to update the README with this pull request? That would be great.

File: src/jquery.ime.js [hunk 3]
```diff
@@ -270,15 +333,18 @@
 		/**
 		 * Set the current input method
 		 * @param {string} inputmethodId
+		 * @fires imeLanguageChange
```
Comment (Nikerabbit) [inline src/jquery.ime.js]: This should have been imeMethodChange, right @divec ?

**Pull request merged**